Задание к уроку 7:

mnikitin77 opened the pull request:
Создать класс, который может выполнять «тесты». В качестве тестов выступают классы с наборами методов с аннотациями @Test. Для этого у него должен быть статический метод start(), которому в качестве параметра передается или объект типа Class, или имя класса. Из «класса-теста» вначале должен быть запущен метод с аннотацией @BeforeSuite, если такой имеется. Далее запущены методы с аннотациями @Test, а по завершении всех тестов – метод с аннотацией @AfterSuite. К каждому тесту необходимо добавить приоритеты (int числа от 1 до 10), в соответствии с которыми будет выбираться порядок их выполнения. Если приоритет одинаковый, то порядок не имеет значения. Методы с аннотациями @BeforeSuite и @AfterSuite должны присутствовать в единственном экземпляре, иначе необходимо бросить RuntimeException при запуске «тестирования».

batiaev commented on src/com/mvnikitin/lesson2/ProductsDB.java:
```diff
@@ -0,0 +1,186 @@
+
+    private boolean isNumber(String value) {
+        boolean res = false;
+        Pattern p = Pattern.compile("[0-9]+.[0-9]+");
```
^\d+$
\^ - начало строки
\d число
+ повтор последнего символа
$ - конец строки

mnikitin77 replied:
Антон, большое спасибо!

mnikitin77 committed:
Задание к уроку 7:
Создать класс, который может выполнять «тесты». В качестве тестов выступают классы с наборами методов с аннотациями @Test. Для этого у него должен быть статический метод start(), которому в качестве параметра передается или объект типа Class, или имя класса. Из «класса-теста» вначале должен быть запущен метод с аннотацией @BeforeSuite, если такой имеется. Далее запущены методы с аннотациями @Test, а по завершении всех тестов – метод с аннотацией @AfterSuite. К каждому тесту необходимо добавить приоритеты (int числа от 1 до 10), в соответствии с которыми будет выбираться порядок их выполнения. Если приоритет одинаковый, то порядок не имеет значения. Методы с аннотациями @BeforeSuite и @AfterSuite должны присутствовать в единственном экземпляре, иначе необходимо бросить RuntimeException при запуске «тестирования».

mnikitin77 commented:
Улучшу....
